public-ui/src/app/data-browser/views/survey-view/components/survey-question-react.component.tsx

```diff
@@ -0,0 +1,137 @@
+import { Component, Input } from '@angular/core';
+import { BaseReactWrapper } from 'app/data-browser/base-react/base-react.wrapper';
+import { TooltipReactComponent } from 'app/data-browser/components/tooltip/tooltip-react.component';
+import { HighlightReactComponent } from 'app/shared/components/highlight-search/HighlightReactComponent';
+import { reactStyles } from 'app/utils';
+import { ClrIcon } from 'app/utils/clr-icon';
+import { environment } from 'environments/environment';
+import { Configuration, DataBrowserApi } from 'publicGenerated/fetch';
+import * as React from 'react';
+import { SurveyAnswerReactComponent } from './survey-answer-react.component';
+
+
+const api = new DataBrowserApi(new Configuration({ basePath: environment.publicApiUrl }));
+
+const styles = reactStyles({
+    seeAnswers: {
+        fontSize: '14px',
+        cursor: 'pointer',
+        color: '#216fb4',
+    },
+    clIcon: {
+        width: '1.3em',
+        height: '1.3em',
+        color: '#216fb4'
+    }
+});
+
+
+interface Props {
+    isCopeSurvey: boolean;
+    question: any;
+    participantCount: number;
+    surveyConceptId: number;
+    searchTerm: string;
+}
+
+interface State {
+    showAnswers: boolean;
+    questionWithResults: any;
+}
+
+export class SurveyQuestionReactComponent extends React.Component<Props, State> {
+    constructor(props: Props, state: State) {
+        super(props);
+        this.state = {
+            showAnswers: false,
+            questionWithResults: {}
+        };
+    }
+
+    showAnswers(e?) {
+        if (e && e.key !== 'Enter') {
+            return;
+        }
+        this.getAnalysis();
+
+        this.setState({
+            showAnswers: !this.state.showAnswers
+        });
```

Comment:
This will still call the api every time we show/hide the answers, even when the data is already in state. We should check if the data is there before calling:
```suggestion
        if (!this.state.questionWithResults.countAnalysis) {
            // Will need to set showAnswers to true in getAnalysis along with this change
            this.getAnalysis();
        } else {
            this.setState({
                showAnswers: !this.state.showAnswers
            });
        }
```